ambry-commons/src/main/java/com/github/ambry/compression/CompressionMap.java

```diff
@@ -0,0 +1,99 @@
+    CompressionMap newSet = new CompressionMap();
+    for (Compression instance : compressionInstances) {
+      newSet.add(instance);
+    }
+    return newSet;
+  }
+
+  /**
+   * Add an instance of compression.  Every compression instance has a unique name.
+   * Decompression will use the compression algorithm name to find the decompression instance.
+   *
+   * @param compressionInstance The compression instance to add.
+   */
+  public void add(Compression compressionInstance) {
+    Objects.requireNonNull(compressionInstance, "compressionInstance");
+
+    put(compressionInstance.getAlgorithmName(), compressionInstance);
+  }
+
+  /**
+   * Find the decompression instance based on the algorithm name stored in compressedBuffer.
+   * Only portion of the compressed buffer contains the compressed information.
+   * The buffer may contain other structures.
+   *
+   * @param compressedBuffer The compressed buffer generated by the compress() method.
+   * @param compressedBufferOffset Offset in compressedBuffer to start reading.
+   * @param compressedDataSize The size of compressed data in compressedBuffer.
+   * @return The decompressor that can decompress this compressed buffer.  Null if no decompressor found.
+   */
+  public Compression getByCompressedData(byte[] compressedBuffer, int compressedBufferOffset, int compressedDataSize) {
+    String algorithmName = getAlgorithmName(compressedBuffer, compressedBufferOffset, compressedDataSize);
+    return get(algorithmName);
+  }
+
+  /**
+   * Find the name of the algorithm used to compression the specified compressed buffer.
+   *
+   * @param compressedBuffer The compressed buffer generated by the compress() method.
+   * @param compressedBufferOffset Offset in compressedBuffer to start reading.
+   * @param compressedDataSize The size of compressed data in compressedBuffer.
+   * @return Name of the compression algorithm.
+   */
+  public String getAlgorithmName(byte[] compressedBuffer, int compressedBufferOffset, int compressedDataSize) {
+    BaseCompression.verifyCompressedBuffer(compressedBuffer, compressedBufferOffset, compressedDataSize);
+
+    // The first byte is the version. Skip the version to reach the algorithm name.
+    int algorithmNameOffset = compressedBufferOffset + 1;
+    byte algorithmNameLength = compressedBuffer[algorithmNameOffset];
+    return new String(compressedBuffer, algorithmNameOffset + 1, algorithmNameLength, StandardCharsets.UTF_8);
+  }
+  /**
+   * Find the decompression instance based on the algorithm name stored in compressedBuffer.
+   * The entire buffer is dedicated to the compressed data.
+   *
+   *
+   * @param compressedBuffer The compressed buffer generated by the compress() method.
+   * @return The decompressor that can decompress this compressed buffer.  Null if no decompressor found.
+   */
+  public Compression getByCompressedData(byte[] compressedBuffer) {
+    Objects.requireNonNull(compressedBuffer, "compressedBuffer");
+
```

Comment:
rm empty line

Reply:
Regarding the empty lines after verification.  This is a nit and depends on personal style preference.  Some people, including myself, like to group code by role.  I prefer to group all parameter checks near the top of the function, followed by an empty line, and then the core code.  Unless there is a specific reason while an empty line is not recommended between parameter checks and core code, I would like to keep the empty line there.

Reply:
It's not clear however, what the purpose of empty line  itself is, which is to separate logically un-related code known only to the author. A comment, preceded and followed by an empty line, would be a better separator.